src/view/new_block_popup_view.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Blockly Demos: Block Factory
+ *
+ * Copyright 2017 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @fileoverview NewBlockPopupView deals with the UI for creating new blocks and
+ * projects.
+ *
+ * @author celinechoo (Celine Choo)
+ */
+class NewBlockPopupView extends PopupView {
+  constructor() {
+    super();
+
+    /**
+     * Popup contents in HTML format.
+     * @type {string}
+     */
+    this.htmlContents = `
+<div class="bkg"></div>
+<div class="box">
+  <div id="exit">x</div>
+  <header>Create a New Block</header>
```

Comment:
I think the above lines are shared by all three implementations, and the base class `PopupView` will need to know about about them to implement `hide()`.  Consider implementing these container `<div>`s in the superclass.